### ISSUE TITLE
Outrunners and clones portaudio outputs sound to the wrong speakers when using 44100 sample rate

### ISSUE THREAD

**Issue by old-gamer** [mamedev/mame#11007]
### MAME version

0.252 (mame0252) and 0.252 (mame0252-316-ge1cd51fc9be)

### System information

Windows 10 Home, 64-bit, version 22H2, USA, US English
Installed RAM 8.00 GB
AMD A6-5200 APU with Radeon(TM) HD Graphics 2.00 GHz
d3d video output

and another machine
Windows 10 Home, 64-bit, version 22H2, USA, US English
Installed RAM 4.00 GB
Intel(R) Pentium(R) CPU B950 @ 2.10GHz
d3d video output

### INI configuration details

```INI
#
# CORE SOUND OPTIONS
#
samplerate                44100
samples                   1
volume                    0
compressor                1
speaker_report            0

#
# OSD SOUND OPTIONS
#
sound                     portaudio
audio_latency             2

#
# PORTAUDIO OPTIONS
#
pa_api                    "Windows WASAPI"
pa_device                 none
pa_latency                0
```

### Emulated system/software

orunners and clones

### Incorrect behaviour

When using a samplerate of 44100 in conjunction with portaudio and windows wasapi, the game plays sound out of the wrong channels i.e. sounds for player 1 come from monitor B side (right) and player 2 comes from monitor A side (left).

Player 1 is monitor A
Player 2 is monitor B

### Expected behaviour

Sound should come from the correct channels.

### Steps to reproduce

1. start mame with the config settings above
2. run the game and enter service mode (f2)
3. go to SOUND OUTPUT (press 9 until it cycles to it, then f2)
4. use 9 to select a monitor (A or B) and play sound/music.

### Additional details

Also, if I set audio_latency to a value of 1, the sound will change sides when changing from windowed to fullscreen and back.  Odd behavior this, why would it do that when changing screens?  Seems kind of like the current problem with vector hlsl doing different things in different screen modes.  I guess I should note I am using the d3d renderer, I don't know if bgfx has the same effect, but these machines don't like bgfx very much.

**Comment by DavidHaywood** [comment]
This sounds to me more like a port audio issue, or something on your system swapping the channels in specific cases?

I'd be surprised if this was a MAME issue, although MAME's audio handling needs an overhaul to support >2 output speakers etc. anyway.

**Comment by old-gamer** [comment]
Doubt it, on 2 completely different machines this happens (noted in post).  It caught my attention while playing through the television, and initially I thought it was something bad with the HDMI cable or TV and so here we are.  No other game I have does this that I am aware of.  I can't think of any off the top of my head that have a setup like Outrunners does with left channel for player 1 and player 2 is the right channel.

**Comment by rb6502** [comment]
Do -sound dsound or -sound xaudio2 also have the channels "reversed"?

**Comment by b0x-Cub3d** [comment]
Another question to ask is this present with the other System Multi 32 games? The way how the games are setup is that one speaker is for each player, so the stereo setup is actually dual-mono iirc.

**Comment by old-gamer** [comment]
> Do -sound dsound or -sound xaudio2 also have the channels "reversed"?

No, dsound and xaudio2 are fine, just portaudio with 44100 rate is the problem with wasapi.

I'm on mame 0.253 official stable and now the sound comes out of the wrong channels the whole time unless I hold the insert key to fast forward, then it does this odd thing where the wrong side plays slow, then the correct channel starts playing for a couple seconds but at fast forward speed (around 225% for me), then goes back n forth, wrong channel normal speed, correct channel fast speed.

I tried recording a wav file of this for you to listen to, but here is the kicker, it plays normally in the wav file, correct speakers are playing for the correct monitor (left speaker is monitor A and right monitor B don't forget!).

**Comment by rb6502** [comment]
The "System 32 Multi" games all have the dual-mono setup.  Those games are Hard Dunk, Stadium Cross, Title Fight, and of course OutRunners.

The next version of MAME will include a massively updated version of PortAudio, taking it from 2015 to their latest development on GitHub as of March 31, 2023.  There are significant updates to the WASAPI support, so hopefully it will fix this issue.

**Comment by old-gamer** [comment]
> The "System 32 Multi" games all have the dual-mono setup. Those games are Hard Dunk, Stadium Cross, Title Fight, and of course OutRunners.
> 
> The next version of MAME will include a massively updated version of PortAudio, taking it from 2015 to their latest development on GitHub as of March 31, 2023. There are significant updates to the WASAPI support, so hopefully it will fix this issue.

Yeah I saw that the other day in the history.  I had a hunch it might fix the problem too, and thanks for mentioning Stadium Cross because that was the one motorcycle game I tend to forget about that I have on here now.  I'll look at the others too, they are on the external drive.

**Comment by old-gamer** [comment]
Sorry to report that this issue is still present in 0.254 with the updated portaudio.  Holding Insert key is still fast for 2 seconds then slow for 1 second and alternating like before.  Seems like it should play at consistent fast speed for the entirety of the key press.  Below are the notes on exactly what happens with each setting configuration so far.

These settings produce correct speaker output (right from right/left from left):
samplerate                44100
sound                     portaudio
audio_latency             0
pa_api                    "Windows WASAPI"
pa_device                 none
pa_latency                0

These flip the speakers to wrong sides:
samplerate                44100
sound                     portaudio
audio_latency             2
pa_api                    "Windows WASAPI"
pa_device                 none
pa_latency                0

These flip the speakers, and flip them back temporarily if you switch from fullscreen to window vice versa:
samplerate                44100
sound                     portaudio
audio_latency             1
pa_api                    "Windows WASAPI"
pa_device                 none
pa_latency                0

The pa_latency doesn't seem to affect which side is playing which speaker, only the **audio_latency** is making it behave badly on 1 or 2.  I haven't even tried with higher numbers yet, but then I'm aiming for low latency and speed, and despite what I've read, going from 48000 to 44100 does boost performance some for this hunk of junk, and helps with the cv1k and taito_f3 games (along with changing resolution to 1280x720 in windows from 1600x900).
